subsys/net/ip/nbr.h

```diff
@@ -92,6 +92,9 @@ struct net_nbr_table {
 	/** Link to a neighbor pool */
 	struct net_nbr *nbr;
 
+	/** Max number of neighbors in the pool */
+	const u16_t nbr_count;
+
```

Comment:
Sure, ordering does not really matter currently in the struct but if we need to add more items then it is useful to have the count as a last member.